iceoryx_hoofs/include/iceoryx_hoofs/internal/design_pattern/static_lifetime_guard.inl

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_DESIGN_STATIC_LIFETIME_GUARD_INL
+#define IOX_HOOFS_DESIGN_STATIC_LIFETIME_GUARD_INL
+
+#include "iceoryx_hoofs/design_pattern/static_lifetime_guard.hpp"
+
+#include <atomic>
```

Comment:
Maybe. But inl. is similar to .cpp in intention (but not visibility) and then it should explicitly include what it uses. 
Of course it will just not compile if the header does not need it anymore but the implementation does so one could remove it here.

The bad case is when one only includes something in the .hpp if it is not needed to avoid including it in .inl.

I am not sure what about our current policy here @elBoberido ?

Reply:
I think it is ok here to remove it, since no new atomics are introduced in the implementation, all of them are defined in the hpp. Removed.